minorminer/layout/layout.py

```diff
@@ -161,19 +161,18 @@ def _graph_distance_matrix(G,
     if disconnected_distance is None:
         disconnected_distance = len(G)
 
-    return np.array(
-        [
-            [V.get(v, disconnected_distance) for v in sorted(G)]
-                 for u, V in all_pairs_shortest_path_length
-        ]
-    )
+    dist_list = [
+        (u, [V.get(v, disconnected_distance) for v in sorted(G)])
+         for u, V in nx.all_pairs_shortest_path_length(G)
```

Comment:
why not `all_pairs_shortest_path_length`?

Reply:
Because in case `G` is disconnected, `nx.all_pairs_shortest_path_length` doesn't return a value for nodes in different connected components. So, in that case, if we use `V.get(v)`, we'll end up with None values in the returned array, whereas for the purposes of the algorithm, it makes more sense to have a distance larger than the max distance in connected components.